File: Syncfusion_Sample/Windows/Tools.Windows/Samples/2.0/Editors Package/Container controls/Popup in Popup/VB/PopupUserControl.vb
#Region "Copyright Syncfusion Inc. 2001 - 2013"
'
'  Copyright Syncfusion Inc. 2001 - 2013. All rights reserved.
'
'  Use of this code is subject to the terms of our license.
'  A copy of the current license can be obtained at any time by e-mailing
'  licensing@syncfusion.com. Any infringement will be prosecuted under
'  applicable laws. 
'
#End Region


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Windows.Forms

Imports Syncfusion.Windows.Forms
Imports Syncfusion.Windows.Forms.Tools

Namespace PopupControls
	''' <summary>
	''' Summary description for PopupUserControl.
	''' </summary>
  Public Class PopupUserControl : Inherits System.Windows.Forms.UserControl
	Private textBox1 As System.Windows.Forms.TextBox
	Private WithEvents button1 As System.Windows.Forms.Button
	Private components As System.ComponentModel.IContainer=Nothing

		Public Sub New()
			' This call is required by the Windows.Forms Form Designer.
			InitializeComponent()

			' TODO: Add any initialization after the InitForm call

		End Sub

		''' <summary> 
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not components Is Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		#Region "Component Designer generated code"
		''' <summary> 
		''' Required method for Designer support - do not modify 
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.textBox1 = New System.Windows.Forms.TextBox()
			Me.button1 = New System.Windows.Forms.Button()
			Me.SuspendLayout()
			' 
			' textBox1
			' 
			Me.textBox1.Location = New System.Drawing.Point(16, 16)
			Me.textBox1.Multiline = True
			Me.textBox1.Name = "textBox1"
			Me.textBox1.Size = New System.Drawing.Size(160, 128)
			Me.textBox1.TabIndex = 0
			Me.textBox1.Text = "<<<  Enter Text >>>"
			' 
			' button1
			' 
			Me.button1.FlatStyle = System.Windows.Forms.FlatStyle.System
			Me.button1.Location = New System.Drawing.Point(24, 184)
			Me.button1.Name = "button1"
			Me.button1.TabIndex = 1
			Me.button1.Text = "OK"
'			Me.button1.Click += New System.EventHandler(Me.button1_Click);
			' 
			' PopupUserControl
			' 
			Me.BackColor = System.Drawing.Color.FromArgb((CByte(192)), (CByte(200)), (CByte(219)))
			Me.Controls.AddRange(New System.Windows.Forms.Control() { Me.button1, Me.textBox1})
			Me.Name = "PopupUserControl"
			Me.Size = New System.Drawing.Size(224, 216)
			Me.ResumeLayout(False)

		End Sub
		#End Region

	  Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button1.Click
		' Find my parent PopupControlContainer:
		If TypeOf Me.FindForm() Is PopupHost Then
			Dim pcc As PopupControlContainer = Nothing
			' Find the PopupControlContainer
			Dim c As Control
			For Each c In Me.FindForm().Controls
				If TypeOf c Is PopupControlContainer Then
					pcc = CType(IIf(TypeOf c Is PopupControlContainer, c, Nothing), PopupControlContainer)
					Exit For
				End If
			Next c
			' Need to do this only if the PopupControlContainer is currently showing
			If Not pcc Is Nothing AndAlso pcc.IsShowing() Then
				pcc.HidePopup(PopupCloseType.Done)
			End If
		End If
	  End Sub
  End Class
End Namespace

File: Syncfusion_Sample/Windows/Tools.Windows/Samples/2.0/Editors Package/Container controls/Popup in Popup/CS/PopupUserControl.cs
#region Copyright Syncfusion Inc. 2001 - 2013
//
//  Copyright Syncfusion Inc. 2001 - 2013. All rights reserved.
//
//  Use of this code is subject to the terms of our license.
//  A copy of the current license can be obtained at any time by e-mailing
//  licensing@syncfusion.com. Any infringement will be prosecuted under
//  applicable laws. 
//
#endregion

using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Windows.Forms;

using Syncfusion.Windows.Forms;
using Syncfusion.Windows.Forms.Tools;

namespace PopupControls
{
	/// <summary>
	/// Summary description for PopupUserControl.
	/// </summary>
  public class PopupUserControl : System.Windows.Forms.UserControl
  {
    private System.Windows.Forms.TextBox textBox1;
    private System.Windows.Forms.Button button1;
    private System.ComponentModel.IContainer components=null;
    
		public PopupUserControl()
		{
			// This call is required by the Windows.Forms Form Designer.
			InitializeComponent();

			// TODO: Add any initialization after the InitForm call

		}

		/// <summary> 
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Component Designer generated code
		/// <summary> 
		/// Required method for Designer support - do not modify 
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.textBox1 = new System.Windows.Forms.TextBox();
			this.button1 = new System.Windows.Forms.Button();
			this.SuspendLayout();
			// 
			// textBox1
			// 
			this.textBox1.Location = new System.Drawing.Point(16, 16);
			this.textBox1.Multiline = true;
			this.textBox1.Name = "textBox1";
			this.textBox1.Size = new System.Drawing.Size(160, 128);
			this.textBox1.TabIndex = 0;
			this.textBox1.Text = "<<<  Enter Text >>>";
			// 
			// button1
			// 
			this.button1.FlatStyle = System.Windows.Forms.FlatStyle.System;
			this.button1.Location = new System.Drawing.Point(24, 184);
			this.button1.Name = "button1";
			this.button1.TabIndex = 1;
			this.button1.Text = "OK";
			this.button1.Click += new System.EventHandler(this.button1_Click);
			// 
			// PopupUserControl
			// 
			this.BackColor = System.Drawing.Color.FromArgb(((System.Byte)(192)), ((System.Byte)(200)), ((System.Byte)(219)));
			this.Controls.AddRange(new System.Windows.Forms.Control[] {
																		  this.button1,
																		  this.textBox1});
			this.Name = "PopupUserControl";
			this.Size = new System.Drawing.Size(224, 216);
			this.ResumeLayout(false);

		}
		#endregion

  	private void button1_Click(object sender, System.EventArgs e)
	  {
		// Find my parent PopupControlContainer:
		if(this.FindForm() is PopupHost)
		{
			PopupControlContainer pcc = null;
			// Find the PopupControlContainer
			foreach(Control c in this.FindForm().Controls)
			{
				if(c is PopupControlContainer)
				{
					pcc = c as PopupControlContainer;
					break;
				}
			}
			// Need to do this only if the PopupControlContainer is currently showing
			if(pcc != null && pcc.IsShowing())
			{
				pcc.HidePopup(PopupCloseType.Done);
			}
		}
	}
  }
}
